src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1846 @@
+
+    def forward(self, tokens: torch.Tensor, num_steps: int, prev_steps: int) -> torch.Tensor:
+        assert tokens.ndim == 2  # x is (batch_size, num_timesteps)
+        output = torch.zeros(*tokens.size(), self.embedding_dim, device=tokens.device)
+        for slicer, emb in zip(self.slicers, self.embedding_tables):
+            s = slicer.compute_slice(num_steps, prev_steps)
+            output[:, s] = emb(tokens[:, s])
+        return output
+
+
+class IrisEncoder(nn.Module):
+    def __init__(self, config) -> None:
+        super().__init__()
+        self.config = config
+        self.num_resolutions = len(config.ch_mult)
+        timestep_embedding_channels = 0  # timestep embedding #channels
+
+        # downsampling
+        self.conv_in = nn.Conv2d(config.in_channels, config.ch, kernel_size=3, stride=1, padding=1)
+
+        curr_res = config.resolution
+        in_ch_mult = (1,) + tuple(config.ch_mult)
+        self.down = nn.ModuleList()
+        for i_level in range(self.num_resolutions):
+            block = nn.ModuleList()
+            attn = nn.ModuleList()
+            block_in = config.ch * in_ch_mult[i_level]
+            block_out = config.ch * config.ch_mult[i_level]
+            for _ in range(self.config.num_res_blocks):
+                block.append(
+                    IrisResnetBlock(
+                        in_channels=block_in,
+                        out_channels=block_out,
+                        timestep_embedding_channels=timestep_embedding_channels,
+                        dropout=config.dropout,
+                    )
+                )
+                block_in = block_out
+                if curr_res in config.attn_resolutions:
+                    attn.append(IrisAttnBlock(block_in))
+            down = nn.Module()
+            down.block = block
+            down.attn = attn
+            if i_level != self.num_resolutions - 1:
+                down.downsample = IrisDownsample(block_in, with_conv=True)
+                curr_res = curr_res // 2
+            self.down.append(down)
+
+        # middle
+        self.mid = nn.Module()
+        self.mid.block_1 = IrisResnetBlock(
+            in_channels=block_in,
+            out_channels=block_in,
+            timestep_embedding_channels=timestep_embedding_channels,
+            dropout=config.dropout,
+        )
+        self.mid.attn_1 = IrisAttnBlock(block_in)
+        self.mid.block_2 = IrisResnetBlock(
+            in_channels=block_in,
+            out_channels=block_in,
+            timestep_embedding_channels=timestep_embedding_channels,
+            dropout=config.dropout,
+        )
+
+        # end
+        self.norm_out = nn.GroupNorm(num_groups=32, num_channels=block_in, eps=1e-6, affine=True)
+        self.conv_out = nn.Conv2d(block_in, config.z_channels, kernel_size=3, stride=1, padding=1)
+
+    def forward(
+        self, x: torch.Tensor, output_hidden_states: bool = False, output_attentions: bool = False
```

Comment:
encoder and deocder should have `output_attentions` and `output_hidden_states` unset and default to the config